### ISSUE TITLE
What's the difference between this and Twin.macro? What are the pros and cons of using each?

### ISSUE THREAD

**Issue by lumenwrites** [MathiasGilson/Tailwind-Styled-Component#65]
This library looks amazing, and seems like it does exactly what I need - enabling me to use Tailwind with React without turning my JSX into an unreadable mess.

Since I'm new to Tailwind, I'm still trying to find the best way to use it. Can you help me figure out the difference between this library and [Twin.macro](https://github.com/ben-rogerson/twin.macro)? What are the pros and cons of using each?

**Comment by FranciscoKloganB** [comment]
Would like to know more about this too. My initial reaction is that `Twin.Macro` is a more complete package (supports more and more complex use cases), but also has more boilerplate when writing code. Then again, I am no expert.

**Comment by MathiasGilson** [comment]
Honnestly the main difference was the syntax of Twin.Macro that was different before and the inheritance that seems to be supported now. I didn't test the last version yet but Twin.Macro seems better maintained too...

**Comment by FranciscoKloganB** [comment]
@MathiasGilson I would like to add another question. :thinking: 

I have been using `tailwind-styled-components` on Next 12 with `SWR` compiler without custom `.babelrc` without issue; However, I assume you did not make a port to work with Rust. So how is this working? :sweat_smile:

**Comment by MathiasGilson** [comment]
It runs at runtime which is less than ideal

**Comment by max-sym** [comment]
twin.macro doesn't allow you to use variables well AFAIK. This package is very simple, doesn't run in the browser and is much lighter.

**Comment by FranciscoKloganB** [comment]
Yes, I am also having lots of trouble using `twin.macro`, been trying to migrate for a week now and I am hitting all kinds of roadblocks. From unclear setup documentation to being forced to use `babel` instead of `swc`.

Sure, performance gains may be nice, if most things are done at compile-time, but I find myself not knowing how to do some things with ease... readability is already being hurt and there is a use-case, in particular, I am not being able to implement. Might be due to inexperience with the package... Do not want to be unfair to `twin.macro` since many people do vouch for its powerful nature.

So from my limited experience, if readability and maintainability is more important than performance, I would definitely stick with `tailwind-styled-components`. I wonder if the run-time overhead of `tailwind-styled-components` is that critical either way. Can you comment on this @MathiasGilson?

Finally, one thing I now value a lot with `tailwind-styled-components` is that it does not force you to be on `tailwind-styled-components` all the way. You can still go with regular `className="things"`, whereas `twin.macro` forces you into the ecosystem all the way. Since regular `className="things"` will generate zero styling, because we import base styles, utilities and components from twin instead of tailwind.

**Comment by ivvv111** [comment]
@FranciscoKloganB @MathiasGilson Hello! Can you explain again what twin.macro does in compile time and why it has better performance? 
Also as I understand twin.macro generates absolutely new classes, it just takes rules from tailwind's classes. So It generates much more css code. tailwind-styled-components uses classes generated by tailwind so it's more more effective for big projects.

**Comment by jahirfiquitiva** [comment]
What I can tell is `tailwind-styled-components` is just an easy (and different) way to write your tailwind classes, but with the same results as you would expect when writing them in the `className` tag.

Example:

| **`tailwind-styled-components`** | **`tailwindcss`** |
|--------|--------|
| <img width="344" alt="Shot 2023-04-10 at 17 38 57@2x" src="https://user-images.githubusercontent.com/10360816/231012329-019e6f68-3f33-4011-9eb4-d85c0fca59bf.png"> | <img width="545" alt="Shot 2023-04-10 at 17 39 35@2x" src="https://user-images.githubusercontent.com/10360816/231012383-5a0a19a9-90e4-4e11-bc5c-d6a02a57b1f7.png"> |

Output HTML in both cases:
```html
<section class="flex flex-col px-14 md:px-0 ..." ... />
```

Whilst with `twin.macro` you (a) need a styled-components library (styled-components, emotion, stitches) and (b) it transforms your tailwind classes into actual CSS with new CSS classes.

So, for the example above:

```typescript
import tw from 'twin.macro';

export const Section = tw.section`
  flex
  flex-col
  px-14
  tablet-md:px-0
`;
```

**CSS output**
```scss
.Section-abc-123 {
  display: flex;
  flex-direction: column;
  padding-left: 14px;
  padding-right: 14px;
}

@media (min-width: 768px) {
  .Section-abc-123 {
    padding-left: 0;
    padding-right: 0;
  }
}
```

**HTML output**
```html
<section class="Section-abc-123" ... />
```

> **Disclaimer:** the properties above will depend on your tailwind configuration
